Lombiq.Tests.UI/SecurityScanning/SecurityScanningUITestContextExtensions.cs

```diff
@@ -116,32 +154,47 @@ public static class SecurityScanningUITestContextExtensions
-
-        var configuration = context.Configuration.SecurityScanningConfiguration;
-
-        SecurityScanResult result = null;
-        try
+        var configuration = context.Configuration.SecurityScanningConfiguration ?? new SecurityScanningConfiguration();
+        async Task RunAndAssertSecurityScanInnerAsync(Uri startUri, Action<SecurityScanConfiguration> configure)
         {
-            result = await context.RunSecurityScanAsync(automationFrameworkYamlPath, configure);
+            SecurityScanResult result = null;
+            try
+            {
+                result = await context.RunSecurityScanAsync(automationFrameworkYamlPath, configure, startUri);
 
-            if (assertSecurityScanResult != null) assertSecurityScanResult(result.SarifLog);
-            else configuration?.AssertSecurityScanResult(context, result.SarifLog);
+                if (assertSecurityScanResult != null) assertSecurityScanResult(result.SarifLog);
+                else configuration.AssertSecurityScanResult(context, result.SarifLog);
 
-            if (configuration.CreateReportAlways) context.AppendDirectoryToFailureDump(result.ReportsDirectoryPath);
-        }
-        catch (Exception ex)
-        {
-            if (result != null) context.AppendDirectoryToFailureDump(result.ReportsDirectoryPath);
-            throw new SecurityScanningAssertionException(ex);
+                if (configuration.CreateReportAlways)
+                {
+                    context.AppendDirectoryToFailureDump(result.ReportsDirectoryPath);
+                }
+            }
+            catch (Exception ex)
+            {
+                if (result != null) context.AppendDirectoryToFailureDump(result.ReportsDirectoryPath);
+                throw new SecurityScanningAssertionException(ex);
+            }
         }
 
-        // Clear app logs before app log assertion is restored.
-        context.ClearLogs();
-        context.Configuration.AssertAppLogsAsync = assertAppLogsAsync;
+        await RunAndAssertSecurityScanInnerAsync(startUri: null, configure);
+
+        // Verify that error page handling also works by visiting a known error page with no logging.
+        var errorUrl = context.GetAbsoluteUri(context.GetRelativeUrlOfAction<ErrorController>(controller => controller.Index()));
+        await context.DoWithoutAppLogAssertionAsync(() => RunAndAssertSecurityScanInnerAsync(
+            startUri: errorUrl,
+            scanConfiguration =>
+            {
+                configure?.Invoke(scanConfiguration);
+
+                // Configure this scan to only visit the target page.
+                scanConfiguration.ModifyZapPlan(plan =>
+                {
+                    plan.SetSpiderParameter("maxDepth", 1.ToTechnicalString());
+                    plan.SetSpiderParameter("maxChildren", 2.ToTechnicalString());
+                    plan.SetActiveScanParameter("recurse", "false");
+                });
+            }));
     }
```

Comment:
![giphy](https://github.com/Lombiq/UI-Testing-Toolbox/assets/4007293/3ad402bb-a103-40fe-af17-c2ca8681b537)
but "temporary log filter" and "requester" aren't among them :(

I couldn't find a schema or complete documentation for this yaml configuration, I hunted the above settings together the from various sources. I don't know if it's possible to turn off recursion only for a specific page. Without that, if I just added the error page as an additional start URL then the scans will take twice as long because it will try to probe the same site (access all the menu links) from two largely equivalent starting points. Granted, I don't know a lot about ZAP so maybe I'm just missing something...

Reply:
Sorry, to clarify: 

- Log filter: you can set the log level in the appsettings to disable logging for just one class. You can also do this from code with [`WithFilter`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.filterloggerfactoryextensions.withfilter), see [sample](https://gunnarpeipman.com/aspnet-core-log-filtering/). But: it now dawned on me that the logging is not actually happening in the controller, so I'm not sure. Please check out if there's some straightforward way to prevent that exception from being logged. If not, then we'll revisit this together with https://github.com/Lombiq/UI-Testing-Toolbox/issues/206 and the current code can stay.
- You can add the [request_o_r job](https://www.zaproxy.org/docs/desktop/addons/automation-framework/job-requestor/) to a YAML to instruct ZAP to visit a URL, any time in the plan. There's also `AddRequestor()` for this.